sdk/nodejs/runtime/rpc.ts

```diff
@@ -206,7 +196,7 @@ export function resolveProperties(
     for (const k of Object.keys(resolvers)) {
         if (!allProps.hasOwnProperty(k)) {
             const resolve = resolvers[k];
-            resolve(isDryRun() ? unknown : undefined, !isDryRun(), false);
+            resolve(undefined, !isDryRun(), false);
         }
     }
 }
```

Comment:
overall this is 10x better.  still have an open thought that we could just avoid everything related to setting 'known ness' bits here as it feels computable from the actual value we get.  i.e. my preference would be to convert the two resolve calls to:

`resolve(value, isSecret);`  and
`resolve(isDryRun() ? unknown : undefined, /*isSecret: false*/)`

i personally feel like that would be clearer and with less chance of error.  

however, i defer to you here.  the overall improvements are already well worth it.

Reply:
I am not going to refactor the resolvers with this PR. It is large enough as-is.

Reply:
👍